### PR TITLE
Moved provider_info_endpoint from extensions to oauth

### PR DESCRIPTION
**Pull request by tpazderka**
Part of #638

- [x] Any changes relevant to users are recorded in the `CHANGELOG.md`.
- [ ] The documentation has been updated, if necessary.
---

### CODE REVIEW

File: src/oic/oic/provider.py [hunk 1]
```diff
@@ -1786,11 +1777,9 @@ def delete_registration(self, authn, request, **kwargs):
 
     def create_providerinfo(self, pcr_class=None, setup=None):
         """
-        Dynamically create the provider info response.
+        Overriden to use the proper message class.
```
Comment (schlenk) [inline src/oic/oic/provider.py]: 'Overridden'

File: src/oic/oauth2/provider.py [hunk 2]
```diff
@@ -317,6 +363,145 @@ def _verify_redirect_uri(self, areq):
...
+
+        _log_info("@providerinfo_endpoint")
+        try:
+            _response = self.create_providerinfo()
+            msg = "provider_info_response: {}"
+            _log_info(msg.format(sanitize(_response.to_dict())))
+            if self.events:
+                self.events.store("Protocol response", _response)
+
+            headers = [("Cache-Control", "no-store"), ("x-ffo", "bar")]
```
Comment (schlenk) [inline src/oic/oauth2/provider.py]: Any idea what the x-ffo header is for?

File: src/oic/oic/provider.py (hunk 1 above)
Comment (tpazderka) [reply src/oic/oic/provider.py]: Fixed.

File: src/oic/oauth2/provider.py (hunk 2 above)
Comment (tpazderka) [reply src/oic/oauth2/provider.py]: Hm, could stand for `X-Forwarded-For`? But the setting (and the header itself) are nonsense.

I am for removing it as it is not in a list of standard and common non-standard headers.

**Pull request merged**